snap/quota/quota.go

```diff
@@ -171,7 +175,7 @@ func (grp *Group) GetQuotaResources() Resources {
 		if grp.JournalLimit.Size != 0 {
 			resourcesBuilder.WithJournalSize(grp.JournalLimit.Size)
 		}
-		if grp.JournalLimit.RateCount != 0 && grp.JournalLimit.RatePeriod != 0 {
```

Comment:
**If I'm understanding the issue correctly**, it might help to add a comment here:

> We cannot just check for RateCount and RatePeriod and call WithJournalRate() only if both are non-zero, because not calling `WithJournalRate()` causes the system's default rate count and rate period to be used; what we we really want here is to completely disable the limits.

Reply:
You are entirely correct in this missing comment! Really nice suggestion, and you understand correctly, sorry for my cryptic explanations so far!